apps/data-pipeline/sample-program-scraper/src/index.ts

```diff
@@ -0,0 +1,590 @@
+    }
+
+    // Year might be OUTSIDE table (Dance programs, etc)
+    if (!foundYearHeader && currentCurriculum.length > 0) {
+      // Look for heading before the table
+      const prevHeading = $table.prevAll("h4, h5, h6, p").first();
+      let yearText: AcademicYearType = "Freshman"; // Default fallback specifically for dance programs
+
+      if (prevHeading.length) {
+        const headingText = prevHeading.text().trim();
+        // Extract year from headings like "Sample Program for Freshmen"
+        if (headingText.match(/freshmen|freshman/i)) {
+          yearText = "Freshman";
+        } else if (headingText.match(/sophomore/i)) {
+          yearText = "Sophomore";
+        } else if (headingText.match(/junior/i)) {
+          yearText = "Junior";
+        } else if (headingText.match(/senior/i)) {
+          yearText = "Senior";
+        } else {
+          yearText = headingText as AcademicYearType;
+        }
+      }
+
+      sampleYears.push({ year: yearText, curriculum: currentCurriculum });
+      logger.info(`Found year outside table: "${yearText}"`);
+    }
+
+    return sampleYears;
+  };
+
+  // CASE 1: Single Table - No Label Needed
+
+  const variations: Array<{
+    label?: string;
+    sampleProgram: SampleProgramEntry[];
+    variationNotes: string[];
+  }> = [];
+
+  if (tableCount === 1) {
+    logger.info("Parsing single table (no variations)");
+
+    const $table = allTables.first();
+    const sampleYears = parseTable($table);
+
+    if (sampleYears.length > 0) {
+      const transformedResult = transformToTermStructure(sampleYears);
+      // For single-table programs, use comprehensive note parsing
+      const variationNotes: string[] = [];
+
+      // Look for "NOTES:" heading with paragraphs
+      sampleProgramContainer.find("p").each((_i, p_el) => {
+        const pText = $(p_el).text().trim();
+        if (pText.match(/^NOTES\s*:\s*/i)) {
+          const remainingText = pText.replace(/^NOTES\s*:\s*/i, "").trim();
+          if (remainingText.length > 0) {
+            variationNotes.push(remainingText);
+          }
+          let currentElement = $(p_el).next();
+          while (currentElement.length && !currentElement.is("h1, h2, h3, h4, h5, h6, table")) {
+            if (currentElement.is("p")) {
+              const paragraphContent = currentElement.text().trim();
+              if (paragraphContent.length > 0 && !paragraphContent.match(/^NOTES\s*:\s*/i)) {
+                variationNotes.push(paragraphContent);
+              }
+            } else if (currentElement.is("ol")) {
+              currentElement.find("li").each((_k, li_el) => {
+                variationNotes.push($(li_el).text().trim());
+              });
+              break;
+            }
+            currentElement = currentElement.next();
+          }
+        }
+      });
+
+      // Look for <dl class="sc_footnotes">
+      if (variationNotes.length === 0) {
```

Comment:
Comment here to make it clear that this is a second-priority fallback (it is, right?)